### PR TITLE
Buzzer implementation

### PR DESCRIPTION
**Pull request by Coscolin**
Added buzzer alarm and configuration in menu.

Alarm: ON / OFF
Repeat: ONCE / EVERY
Each: 5s / 10s / 15s / 30s / 1m / 2m / 5m
Tone: HIGH / MED. / LOW
Span: SHORT / MED / LONG

Inside menu buzzer is silenced.

Tested and work correctly.

TODO: save changes in memory

### CODE REVIEW

**Comment by Coscolin** [comment]
@melkati , the file [WiFiMenu.h] it's not necesary to commit.

It's for another implementation and is still in develpment.

Don't import it to master.

I'm sorry, I'm still learning GitHub metodology.

**Comment by melkati** [comment]
> Alarm: ON / OFF
> Repeat: ONCE / EVERY
> Each: 5s / 10s / 15s / 30s / 1m / 2m / 5m
> Tone: HIGH / MED. / LOW
> Span: SHORT / MED / LONG

What's Repeat, Each, Span, intended for?

I will leave some comments in the PR as I review it.

I see this PR implements buffer control ("hardware" control) and also menu modification. Please separate both in two PR for enhanced manageability.

Thank you! 😀

**Comment by melkati** [comment]
As including this feature is not easy and touches many things in CO2 Gadget, I created a new branch (feature-buzzer) for it, so we can merge it once everything is complete.

**Review by melkati — CHANGES_REQUESTED** [review]

File: CO2_Gadget_Buzzer.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*****************************************************************************************************/
+/*********                                                                                   *********/
+/*********                                   SETUP BUZZER                                    *********/
+/*********                                                                                   *********/
+/*****************************************************************************************************/
+
+#define BUZZZER_PIN  13 // ESP32 pin GPIO13 connected to piezo buzzer
+
+int lowestco2=10000;
```
[the remark this answers is not in the document]
Comment (Coscolin) [reply CO2_Gadget_Buzzer.h:9]: Not necessary. It was a failed previous way to implement repeat buzzer only once.

File: CO2_Gadget_Buzzer.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+int highestco2=0;
+
+bool downOrangeRange = true;
+bool downRedRange = true;
+
+
+void buzzerLoop(){
```
[the remark this answers is not in the document]
Comment (Coscolin) [reply CO2_Gadget_Buzzer.h:16]: Only block during sound, about 0.3sec to 1sec.
To non blocking strategy we can use ticker library:
https://techtutorialsx.com/2021/08/07/esp32-ticker-library/

File: CO2_Gadget_Buzzer.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    if(!activeAlarm)        
+        return;
+
+    if(inMenu){  // Inside Menu stop BEEPING
+        noTone(BUZZZER_PIN);
+        return;
+    }
+
+    if ((millis() - lastTimeAlarmBeep >= timeBetweenAlarmBeep * 1000) || (lastTimeAlarmBeep == 0)) {
```
[the remark this answers is not in the document]
Comment (Coscolin) [reply CO2_Gadget_Buzzer.h:26]: It's a good idea. Done.

File: CO2_Gadget_Menu.h [hunk 4]
```diff
@@ -744,6 +744,43 @@ MENU(outputsConfigMenu, "Outputs Config", doNothing, noEvent, wrapStyle
   ,SUBMENU(outputsModeMenu)
   ,EXIT("<Back"));
 
+  #ifdef SUPPORT_BUZZER
+TOGGLE(activeAlarm, activeAlarmOffMenu, "Alarm: ", doNothing, noEvent, wrapStyle
```
[the remark this answers is not in the document]
Comment (Coscolin) [reply CO2_Gadget_Menu.h]: I have changed Alarm to Buzzer but I don't knowc how to divide PR.

**Comment by Coscolin** [comment]
> > Alarm: ON / OFF
> > Repeat: ONCE / EVERY
> > Each: 5s / 10s / 15s / 30s / 1m / 2m / 5m
> > Tone: HIGH / MED. / LOW
> > Span: SHORT / MED / LONG
> 
> What's Repeat, Each, Span, intended for?
> 
> I will leave some comments in the PR as I review it.
> 
> I see this PR implements buffer control ("hardware" control) and also menu modification. Please separate both in two PR for enhanced manageability.
> 
> Thank you! 😀

- REPEAT ONCE: Only buzzer one time orange or red CO2 range is raised. Don't buzzer again until CO2 values downs "HYSTERIC" value down orange/red range and raise again.
- REPEAT EVERY: Buzzer all time while CO2 values are upper orange or red range.
- EACH: time between buzzers if repeat is "EVERY"
- TONE: Frecuency of buzzer tone. 
- SPAN: Duration of buzzer tones.

**Comment by melkati** [comment]
Continues in #157 and #153

**Pull request closed without merging**